src/main/java/foodwhere/logic/parser/RFindCommandParser.java

```diff
@@ -21,15 +23,25 @@ public class RFindCommandParser implements Parser<RFindCommand> {
      * @throws ParseException if the user input does not conform the expected format
      */
     public RFindCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
-        if (trimmedArgs.isEmpty()) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, RFindCommand.MESSAGE_USAGE));
+        ArgumentMultimap argMultimap =
+            ArgumentTokenizer.tokenize(args,
+                    CliSyntax.PREFIX_NAME,
+                    CliSyntax.PREFIX_TAG);
+
+        if ((!arePrefixesPresent(argMultimap, CliSyntax.PREFIX_NAME)
+                && !arePrefixesPresent(argMultimap, CliSyntax.PREFIX_TAG))
+            || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, RFindCommand.MESSAGE_USAGE));
         }
 
-        String[] nameKeywords = trimmedArgs.split("\\s+");
+        Set<Name> nameList = ParserUtil.parseNameList(argMultimap.getAllValues(CliSyntax.PREFIX_NAME));
```

Comment:
I think `nameSet` might be a better name for this variable - same goes for `tagSet` and in SFindCommandParser.

Also, I would like to ask, why are the tokens to match names stored as Name and tokens to match tags stored as Tag, instead of strings?

Reply:
Can, I'll update the variable names. Since users would be searching within a specified field (name or tag), their input should be a valid name or tag. Also, this would allow more specific errors to be thrown, if for example name entered is invalid, the error message would be that names should be alphanumeric. If stored as strings, there would not be any error messages displayed, but shows no results listed instead.